core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2125,16 +2128,23 @@ private long startOffsetDuringInitialization(long partitionDataStartOffset) thro
         if (partitionDataStartOffset != PartitionFactory.UNINITIALIZED_START_OFFSET) {
             return partitionDataStartOffset;
         }
-        GroupConfig.ShareGroupAutoOffsetReset offsetResetStrategy;
+        ShareGroupAutoOffsetResetStrategy offsetResetStrategy;
         if (groupConfigManager.groupConfig(groupId).isPresent()) {
             offsetResetStrategy = groupConfigManager.groupConfig(groupId).get().shareAutoOffsetReset();
         } else {
             offsetResetStrategy = GroupConfig.defaultShareAutoOffsetReset();
         }
 
-        if (offsetResetStrategy == GroupConfig.ShareGroupAutoOffsetReset.EARLIEST)
+        final long timestamp = offsetResetStrategy.timestamp()
+            .orElseThrow(() -> new NoOffsetForPartitionException(topicIdPartition.topicPartition()));
```

Comment:
This is only needed for BY_DURATION i.e. only in `else` block below. So should we move the code there?